modules/vulcan/src/main/scala/fs2/kafka/vulcan/AvroDeserializer.scala

```diff
@@ -29,13 +31,18 @@ final class AvroDeserializer[A] private[vulcan] (
   )(implicit F: Sync[F]): Deserializer.Record[F, A] =
     codec.schema match {
       case Right(schema) =>
-        val createDeserializer: Boolean => F[Deserializer[F, A]] =
-          settings.createAvroDeserializer(_).map { deserializer =>
-            Deserializer.instance { (topic, _, bytes) =>
-              F.suspend {
-                codec.decode(deserializer.deserialize(topic, bytes, schema), schema) match {
-                  case Right(a)    => F.pure(a)
-                  case Left(error) => F.raiseError(error.throwable)
+        val createDeserializer: Boolean => F[Deserializer[F, A]] = isKey =>
+          settings.schemaRegistryClient.flatMap { schemaRegistryClient =>
+            settings.createAvroDeserializer(isKey).map { deserializer =>
+              Deserializer.instance { (topic, _, bytes) =>
+                F.suspend {
+                  val writerSchemaId = ByteBuffer.wrap(bytes).getInt(1) // skip magic byte
+                  val writerSchema = schemaRegistryClient.getById(writerSchemaId)
```

Comment:
It would be nice not to have to do this byte-level stuff to get the writer schema, but I don't think the schema registry API gives a way to avoid it.